### ISSUE TITLE
CSTORE fails on .NET CORE 7.0 using 5.1.1 with 'UnexpectedPDUParameter' but works on Framework 4.7.2 using 4.0.8

### ISSUE THREAD

**Issue by acox-belkiwi** [fo-dicom/fo-dicom#1631]
**Describe the bug**
When using ASP.NET CORE 7.0 and making a simple CSTORE of a study, on this study we always get Association Abort **[source: ServiceProvider; reason: UnexpectedPDUParameter]** exception.  If we use .NET framework 4.7.2, the study is sent in with no issues.

Background:  Create a study anonymization process and we are migrating the code base from Framework 4.7.2 to Core 7.0.  This requires us to move from 4.0.8 to 5.1.1.  This is a common issue we are finding on files that never had any issues but now we have this Unexpected PDU Issue. 

In the source below I did try completing the SendAsync directly after each AddRequestAsync (i.e. sending individually) and this worked with no errors.   So seems that when they are all combined it fails.

Appreciate if anyone has any ideas about what the issue could be?  This is a bit of a show stopper for the moment.

I can give more logging if needed.

**To Reproduce**
I created 2 sample projects (one in Framework and one in CORRE) with a basic CStore that replicates the issue.  You run the same DICOM files through it and get the issue. Was not expecting any issue here.  Code is:

Framework 4.7.2 using 4.0.8
```
static void Main(string[] args)
        {
            Console.WriteLine("ASP.NET Framework 4.7.2 Starting...");

            var client = new Dicom.Network.Client.DicomClient("mei.local", 104, false, "MELLF239", "MEI_TR");

            foreach (string f in Directory.EnumerateFiles(@"C:\TEMP\Out"))
            {
                Console.WriteLine("Adding: " + f);
                client.AddRequestAsync(new DicomCStoreRequest(DicomFile.Open(f))).Wait();
            }

            Console.WriteLine("Sending...");
            try
            {
                client.SendAsync().Wait();
            }
            catch (Exception ex)
            {
                Console.WriteLine($"{ex.Message}");
            }

            Console.WriteLine("Done");
            Console.ReadLine();

        }
```

Core 7.0 using 5.1.1
```
Console.WriteLine("ASP.NET CORE 7.0 Starting...");

var client = DicomClientFactory.Create("mei.local", 104, false, "MELLF239", "MEI_TR");

foreach (string f in Directory.EnumerateFiles(@"C:\TEMP\Out"))
{
    Console.WriteLine("Adding: " + f);
    await client.AddRequestAsync(new DicomCStoreRequest(DicomFile.Open(f)));
}

Console.WriteLine("Sending...");
try
{
    await client.SendAsync();
}
catch (Exception ex)
{
    Console.WriteLine($"{ex.Message}");
}

Console.WriteLine("Done");
Console.ReadLine();
```

**Expected behavior**
5.1.1 would work exactly the same as 4.08 but seems something has changed causing this issue.

**Screenshots or test DICOM files**
Results:
4.7.2
![image](https://github.com/fo-dicom/fo-dicom/assets/118498162/9c35f173-1634-4609-b8c6-07a07d96196a)

7.0
![image](https://github.com/fo-dicom/fo-dicom/assets/118498162/79de4612-1f9b-4818-b51d-1d3b160092c8)

**Environment**
Fellow Oak DICOM version: 5.1.1 compared to 4.0.8
OS: Windows
Platform:  .Net CORE 7.0 compared to  .NET Framework 4.7.2

**Comment by acox-belkiwi** [comment]
After some more testing and trying to debug the source of 5.1.1 have not managed to find out why or the cause yet.  So attached is a logfile of the output on an example of what is happening in the main solution and a ZIP of a study that will not load.  

[CStoredFailedLogOutput.txt](https://github.com/fo-dicom/fo-dicom/files/12189657/CStoredFailedLogOutput.txt)
[DICOMFailedStudyUsing5.1.1.zip](https://github.com/fo-dicom/fo-dicom/files/12189658/DICOMFailedStudyUsing5.1.1.zip)

For now my only solution is to use 4.0.8 so have wrapped this in it's own assembly just for the CStore client.  This not ideal as give a performance hit to make it work and limits logging etc.

I would like to use 5.1.1 so if anyone could see what is happening with the example above the that would be helpful.  I will continue investigation as well as really need to have a clean and compatible Core 7.0 solution.

Thanks!

**Comment by amoerie** [comment]
The logs look okay at first glance. Can you also provide logs for the working 4.0.8 code? 
You say that sending every file separately using SendAsync works? Can you confirm that if you call dicomClient.NegotiateAsyncOps(1,1) this has the same effect? 
Could it then be an issue with the receiving server that it does not support Async Ops Invoked > 1, even though in the association handshake it is saying it supports 20 invoked requests?

If the working 4.0.8 logs are identical to to the broken 5.1.1 logs, then the only remaining option that I can see is a bug somewhere at the PDataTFStream level where we write the wrong bytes, but it would honestly surprise me a bit.

**Comment by acox-belkiwi** [comment]
After a lot of evaluation, there is one instance that contains a Transfer Syntax that the Archive we are sending to does not support if not an image thus we get UnexpectedPDUParameter error response.  Related to the SR instance in the example.  Very cryptic from the Archive.

FYI - I also found that in 4.0.8, the issue was there but did not get an exception so assumed it was ok but is was not when we look closer at the final results as was missing this instance.  In 5.1.1 we got the exception.  No fix needed on this on as not a fo-dicom issue but a data related issue.

Thanks for the support!
